rest_server/RestServer.go

```diff
@@ -0,0 +1,269 @@
+package main
+
+// specs from https://github.com/siodb/siodb/blob/master/docs/dev/designs/RestServer.md
+// TODO 1) Create the directory with code and give you build instruction so that you can integrate it (even if not 100% compliant yet)
+//      2) Finish the POST and DELETE implementation
+//      3) Integrate the rest of the requirements you mentioned (log format, etc)
+// TODO: Put chunked WritePayload in function
+// TODO: Put read chunked payload from IOMgr in function
+// TODO: Once cxxman code step 5 in https://github.com/siodb/siodb/blob/master/docs/dev/designs/RestServer.md#post-patch-requests
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+type RestServer struct {
+	Ipv4HTTPPort          uint64
+	Ipv4HTTPSPort         uint64
+	Ipv6HTTPPort          uint64
+	Ipv6HTTPSPort         uint64
+	TLSCertificate        string
+	TLSPrivateKey         string
+	HTTPChunkSize         uint64
+	logFileDestination    string
+	logFileSeverity       string
+	logConsoleDestination string
+	logConsoleSeverity    string
+}
+
+type Config map[string]string
```

Comment:
replace with structure containing validated option values in their actual data types, not strings